### PR TITLE
postgresql: Add version 12.2

### PR DESCRIPTION
**Pull request by haralmha**

### CODE REVIEW

File: var/spack/repos/builtin/packages/postgresql/package.py [hunk 1]
```diff
@@ -19,6 +19,7 @@ class Postgresql(AutotoolsPackage):
     list_url = "http://ftp.postgresql.org/pub/source"
     list_depth = 1
 
+    version('12.2',   sha256='ad1dcc4c4fc500786b745635a9e1eba950195ce20b8913f50345bb7d5369b5de')
```
Comment (michaelkuhn) [inline var/spack/repos/builtin/packages/postgresql/package.py:23]: Why add this version? The most up-to-date one seems to be 14.0 (and there's also 12.8).

Comment (vvolkl) [reply var/spack/repos/builtin/packages/postgresql/package.py:23]: Hi @michaelkuhn, we're doing this to reproduce a build done with another build tool with spack. But @haralmha, while you are at it, can you add the newer versions as well?

Comment (haralmha) [reply var/spack/repos/builtin/packages/postgresql/package.py:23]: Yes, I can do that @vvolkl!

Comment (haralmha) [reply var/spack/repos/builtin/packages/postgresql/package.py:23]: There is a lot of versions between 12.2 and 14.0, so I only added 14.0. Please let me know if you want me to add more of the versions in between

**Review by michaelkuhn — APPROVED** [review]

**Pull request merged**